datascience/tables.py

```diff
@@ -182,62 +184,37 @@ def __getitem__(self, row_indices_or_slice):
             row_slice = slice(row_slice, row_slice+1)
         return Table.from_rows(itertools.chain(self._table.rows[:row_slice.start or 0],
                                                self._table.rows[row_slice.stop:]),
-                               self._table.column_labels)
-
+                               self._table.labels)
 
 class Table(collections.abc.MutableMapping):
-    """A sequence of labeled columns."""
-
-    ##########
-    # Create #
-    ##########
+    """A sequence of string-labeled columns."""
 
-    def __init__(self, columns=None, labels=None,
-                 formatter=_formats.default_formatter):
-        """Create a table from a list of column values.
+    def __init__(self, labels=None, _other=None, formatter=_formats.default_formatter):
+        """Create an empty table with column labels.
 
-        >>> letters = ['a', 'b', 'c', 'z']
-        >>> counts = [9, 3, 3, 1]
-        >>> points = [1, 2, 2, 10]
-        >>> t = Table([letters, counts, points], ['letter', 'count', 'points'])
-        >>> t
+        >>> tiles = Table(['letter', 'count', 'points'])
+        >>> tiles
         letter | count | points
-        a      | 9     | 1
-        b      | 3     | 2
-        c      | 3     | 2
-        z      | 1     | 10
-
-        For other ways to initialize a table, see :func:`Table.empty`,
-        :func:`Table.from_rows`, :func:`Table.from_records`,
-        :func:`Table.read_table`, and :func:`Table.from_columns_dict`.
 
         Args:
-            ``columns`` (list): A list in which each element is another list
-                containing the values for a column in the order the columns
-                appear in ``labels``.
-
-            ``labels`` (list): A list of column labels.
+            ``labels`` (list of strings): The column labels.
 
             ``formatter`` (Formatter): An instance of :class:`Formatter` that
                 formats the columns' values.
-
-        Returns:
-            A new instance of ``Table``.
-
-        Raises:
-            AssertionError:
-                - ``labels`` is specified but ``columns`` is not and vice-versa.
-                - The length of ``labels`` and the length of ``columns`` are
-                  unequal.
         """
         self._columns = collections.OrderedDict()
         self._formats = dict()
         self.formatter = formatter
 
-        # Ensure inputs are properly formed
-        columns = columns if columns is not None else []
-        labels = labels if labels is not None else []
-        assert len(labels) == len(columns), 'label/column number mismatch'
+        if _other is not None:
+            warnings.warn("deprecated", DeprecationWarning)
```

Comment:
FutureWarning looks like a good fit. It's supposed to indicate that
semantics will change in the future. It is shown by default. What do you
think?

On Fri, Jan 29, 2016 at 12:07 AM, Sam Lau <notifications@github.com> wrote:

> In datascience/tables.py
> <https://github.com/data-8/datascience/pull/187#discussion_r51234741>:
>
> >          """
> >          self._columns = collections.OrderedDict()
> >          self._formats = dict()
> >          self.formatter = formatter
> >
> > -        # Ensure inputs are properly formed
> > -        columns = columns if columns is not None else []
> > -        labels = labels if labels is not None else []
> > -        assert len(labels) == len(columns), 'label/column number mismatch'
> > +        if _other is not None:
> > +            warnings.warn("deprecated", DeprecationWarning)
>
> Makes sense, although it seems likely that I wouldn't know a method was
> deprecated until it got removed entirely. At that point, I'd have breaking
> code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/data-8/datascience/pull/187/files#r51234741>.
>